### PR TITLE
Automate / script Helm chart releases

### PR DESCRIPTION
**Pull request by draychev**
This is not yet full automation - but an attempt to write scripts that systematize the work to release a new Helm chart.

Next step would be to execute this via Azure Pipelines.

Even though we talked about proper templating (Jinja2 etc) - the values we need to change seem so trivial - it seems quicker at this point to simply `sed` the files. (Pipe a template through sed, replace a token with the version, save in the actual YAML file)

To achieve that I introduced `Chart-template.yaml` and `values-template.yaml`. Each one of these has `XXVERSIONXX` which `sed` will replace w/ the latest git tag.

### CODE REVIEW

File: helm/release.sh [hunk 1]
```diff
@@ -1,7 +1,25 @@
...
 echo " - running helm package"
-helm package ingress-azure --version 0.4.0
+helm package ingress-azure --version "$GIT_TAG"
+
 echo " - updating helm repo index"
 helm repo index . --url https://azure.github.io/application-gateway-kubernetes-ingress/helm
-echo " - done!"
\ No newline at end of file
+
+echo " - update helm templates"
```
Comment (akshaysngupta) [inline helm/release.sh]: We should generate the Chart/Values before calling `helm package`.

Comment (akshaysngupta) [reply helm/release.sh]: then you don't have to use `./update-index.py`

Comment (draychev) [reply helm/release.sh]: Thanks!

File: helm/index.yaml [hunk 2]
```diff
@@ -1,9 +1,19 @@
 apiVersion: v1
 entries:
   ingress-azure:
+  - apiVersion: v1
```
Comment (akshaysngupta) [inline helm/index.yaml]: can you omit this file ?

File: helm/ingress-azure/values.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@ replicaCount: 1
 
 image:
   repository: mcr.microsoft.com/azure-application-gateway/kubernetes-ingress
-  tag: 0.4.0
+  tag: 0.5.1
```
Comment (akshaysngupta) [inline helm/ingress-azure/values.yaml]: can you omit this file ?

**Pull request merged**